src/snowflake/cli/_plugins/streamlit/streamlit_entity.py

```diff
@@ -28,43 +25,141 @@ def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
     @property
-    def project_root(self) -> Path:
+    def root(self):
         return self._workspace_ctx.project_root
 
     @property
-    def deploy_root(self) -> Path:
-        return self.project_root / "output" / "deploy"
+    def artifacts(self):
+        return self._entity_model.artifacts
 
-    def action_bundle(
-        self,
-        *args,
-        **kwargs,
-    ):
+    @functools.cached_property
+    def _sql_executor(self):
+        return get_sql_executor()
+
+    @functools.cached_property
+    def _conn(self):
+        return self._sql_executor._conn  # noqa
+
+    @property
+    def model(self):
+        return self._entity_model  # noqa
+
+    def action_bundle(self, action_ctx: ActionContext, *args, **kwargs):
         return self.bundle()
 
-    def bundle(self, bundle_root=None):
-        return build_bundle(
-            self.project_root,
-            bundle_root or self.deploy_root,
-            [
-                PathMapping(src=str(artifact))
-                for artifact in self._entity_model.artifacts
-            ],
+    def action_deploy(self, action_ctx: ActionContext, *args, **kwargs):
+        # After adding bundle map- we should use it's mapping here
+
+        query = self.get_deploy_sql()
+        result = self._sql_executor.execute_query(query)
+        return result
```

Comment:
```suggestion
        return self._sql_executor.execute_query(self.get_deploy_sql())
```
To be consistent with the rest of the functions :)